### PR TITLE
add new extensions configuration

### PR DESCRIPTION
**Pull request by jmazzitelli**
* server PR: https://github.com/kiali/kiali/pull/7486

### CODE REVIEW

File: crd-docs/crd/kiali.io_kialis.yaml [hunk 1]
```diff
@@ -588,6 +588,20 @@ spec:
                     description: "When true, Kiali will be in 'view only' mode, allowing the user to view and retrieve management and monitoring data for the service mesh, but not allow the user to modify the service mesh."
                     type: boolean
 
+              extensions:
+                description: |
+                  Defines third-party extensions whose metrics can be integrated into the Kiali UI.
```
Comment (jshaughn) [inline crd-docs/crd/kiali.io_kialis.yaml:593]:
```suggestion
                  Defines third-party extensions whose metrics can be integrated into the Kiali traffic graph.
```

File: crd-docs/crd/kiali.io_kialis.yaml [hunk 2]
```diff
@@ -588,6 +588,20 @@ spec:
...
+                type: array
+                items:
+                  type: object
+                  properties:
+                    enabled:
+                      description: Determines if the extension is to be visualized in the Kiali UI.
```
Comment (jshaughn) [inline crd-docs/crd/kiali.io_kialis.yaml:599]:
```suggestion
                      description: Determines if the Kiali traffic graph should incorporate the extension's metrics.
```

File: crd-docs/crd/kiali.io_kialis.yaml [hunk 3]
```diff
@@ -588,6 +588,20 @@ spec:
...
+                      type: boolean
+                    name:
+                      description: The name that is used to identify the metric timeseries for the extension.
```
Comment (jshaughn) [inline crd-docs/crd/kiali.io_kialis.yaml:602]:
```suggestion
                      description: The name that is used to identify the metric time series for the extension.
```

**Review by jshaughn — COMMENTED** [review]
Some minor text changes.  Should we go ahead and add an `url` attribute?

**Comment by jmazzitelli** [comment]
> Should we go ahead and add an `url` attribute?

Let's wait until we think more about how we are going to support external URLs

**Comment by jshaughn** [comment]
I created an issue for this effort: https://github.com/kiali/kiali/issues/7485

**Review by jshaughn — APPROVED** [review]
Matches KEP, LGTM

**Pull request merged**